### ISSUE TITLE
[NextJS] Add Next 15.1 support

### ISSUE THREAD

**Issue by gperdomor** [nrwl/nx#28586]
### Current Behavior

Next generator generate apps with next 14.2.3 is used

### Expected Behavior

Next generator should generate apps with next ~15.0~ 15.1

### Steps to Reproduce

N/A

### Nx Report

N/A

### Operating System

- [ ] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

**Comment by Coly010** [comment]
@gperdomor thanks for the issue. This is already on our radar. However, given that since Next 15 released there have been a ton of bug fixes that have shipped to Next 15 patch releases, it might be prudent to wait a little bit longer for Next to be stable (https://github.com/vercel/next.js/commits/canary/)

**Comment by armingdev** [comment]
> [@gperdomor](https://github.com/gperdomor) thanks for the issue. This is already on our radar. However, given that since Next 15 released there have been a ton of bug fixes that have shipped to Next 15 patch releases, it might be prudent to wait a little bit longer for Next to be stable (https://github.com/vercel/next.js/commits/canary/)

I agree that we should wait for the more stable version of NextJS 15, but also there are notable fixes in version 14 and NX should push it to the latest version of NextJS 14. (https://github.com/vercel/next.js/releases/tag/v14.2.16)

The current version has vulnerabilities :

GHSA-gp8f-8m3g-qvj9 7.5 Next. js Cache Poisoning
GHSA-g77x-44xx-532m 5.9 Denial of Service condition in Next. js image optimization

**Comment by ryanolson-aumni** [comment]
> [@gperdomor](https://github.com/gperdomor) thanks for the issue. This is already on our radar. However, given that since Next 15 released there have been a ton of bug fixes that have shipped to Next 15 patch releases, it might be prudent to wait a little bit longer for Next to be stable (https://github.com/vercel/next.js/commits/canary/)

It might be prudent, but can't we have the choice to move to Next.js 15? It would be great to add support for it in @nx/next for those that want to experiment with it and start using it.

**Comment by Reza-tm** [comment]
> > [@gperdomor](https://github.com/gperdomor) thanks for the issue. This is already on our radar. However, given that since Next 15 released there have been a ton of bug fixes that have shipped to Next 15 patch releases, it might be prudent to wait a little bit longer for Next to be stable (https://github.com/vercel/next.js/commits/canary/)
> 
> It might be prudent, but can't we have the choice to move to Next.js 15? It would be great to add support for it in @nx/next for those that want to experiment with it and start using it.

Isn't there any solution for this ?

**Comment by gperdomor** [comment]
I updated the issue for Next 15.1 which was recently released... Also @Coly010 is working on a React 19 PR :hey so probably that will be merged before handling this issue...

**Comment by gentle-noah** [comment]
Feeling like Next 15 is stable enough now to enable in NX

**Comment by gentle-noah** [comment]
@Coly010 - do we have a timeline on getting NextJS 15 support?

**Comment by roman-codegeeks** [comment]
Hi guys! Is there any info or roadmap, when NextJS 15 will be added to nx/next ?

**Comment by ZackInfosbd** [comment]
Hi guys! any news about getting NextJs15 support?

**Comment by nolafs** [comment]
Any idea when?

**Comment by laszlo-ratesic** [comment]
Can we get a timeline on this please?

**Comment by gperdomor** [comment]
Folks next 15 support is added in this MR https://github.com/nrwl/nx/pull/29286
